lib/Mojo/Date.pm

```diff
@@ -54,16 +54,29 @@ sub parse {
 sub to_datetime {
 
   # RFC 3339 (1994-11-06T08:49:37Z)
-  my ($s, $m, $h, $day, $month, $year) = gmtime(my $epoch = shift->epoch);
+  my ($s, $m, $h, $day, $month, $year) = _gmftime(shift->epoch);
   my $str = sprintf '%04d-%02d-%02dT%02d:%02d:%02d', $year + 1900, $month + 1,
     $day, $h, $m, $s;
-  return $str . ($epoch =~ /(\.\d+)$/ ? $1 : '') . 'Z';
+  return $str . ($s =~ /(\.\d+)$/ ? $1 : '') . 'Z';
 }
 
+sub _gmftime {
+  my $epoch = shift;
+  if ($epoch =~ /^(-?\d+)(\.\d+)$/) {
+    my $t = ($epoch < 0) ? $1 - 1      : $1;    # floor
+    my $f = ($epoch < 0) ? $epoch - $t : $2;
+    my @t = gmtime($t);
+    $t[0] += $f;
+    return @t;
+  }
+  return gmtime($epoch);
+}
+
+
 sub to_string {
 
   # RFC 7231 (Sun, 06 Nov 1994 08:49:37 GMT)
-  my ($s, $m, $h, $mday, $month, $year, $wday) = gmtime shift->epoch;
+  my ($s, $m, $h, $mday, $month, $year, $wday) = gmtime sprintf('%.0f', shift->epoch);
```

Comment:
Just to make clear, the rounding is important to make sure `gmtime` is called with an integer argument (which is portable). I mentioned that at the start of the "Summary" of this ticket.